Objective-C/Tests/DatabaseTest.m

```diff
@@ -582,6 +582,48 @@ - (void) testSaveDocWithDeletedConflictUsingConcurrencyControl: (int)concurrency
     [self cleanDB];
 }
 
+- (void) testConflictHandlerWithDeletedOldDoc {
+    NSString* docID = @"doc1";
+    [self generateDocumentWithID: docID];
+    AssertEqual([self.db documentWithID: docID].generation, 1u);
+    
+    // keeps new doc(non-deleted)
+    CBLMutableDocument* doc1a = [[self.db documentWithID: docID] toMutable];
+    CBLMutableDocument* doc1b = [[self.db documentWithID: docID] toMutable];
+    
+    [self deleteDocument: doc1a concurrencyControl: kCBLConcurrencyControlLastWriteWins];
+    
+    NSError* error;
+    [doc1b setString: @"value1" forKey: @"key1"];
+    Assert([self.db saveDocument: doc1b
+                 conflictHandler:^BOOL(CBLMutableDocument * cur, CBLDocument * old) {
+                     AssertNil(old);
+                     AssertNotNil(cur);
+                     return YES;
+                 } error: &error]);
+    AssertNil(error);
```

Comment:
Take this out -- on success, the value of `error` is undefined.

Reply:
ok